src/Form/molecules/FilterDropdown.tsx

```diff
@@ -0,0 +1,537 @@
+      const valueString = (element.value).toString(10);
+      return valueString.includes(newValue);
+    }
+
+    if (isListItem(element) && typeof element.value === 'string') {
+      return element.value.includes(newValue);
+    }
+
+    if (typeof element === 'number') {
+      const valueString = element.toString(10);
+      return valueString.includes(newValue);
+    }
+
+    if (typeof element === 'string') {
+      return element.includes(newValue);
+    }
+
+  });
+}
+
+const getResultText = (template: string, visible: number, total: number) => {
+  const newMessage = template.replace('[TOTAL]', `${total}`);
+  return newMessage.replace('[VISIBLE]', `${visible}`);
+}
+
+type IFilterItem = string | ListItem | number;
+
+export type IFilterDropdownValue = string | string[] | number | number[] | ListItem | ListItem[] | null;
+
+interface ListItem {
+  text: string;
+  value: string | number;
+}
+
+type IOpenPos = 'top-left' | 'top-right' | 'bottom-left' | 'bottom-right'
+
+interface IDropOpen {
+  isOpen: boolean,
+  position: IOpenPos,
+}
+interface IFilterDropdown {
+  buttonIcon: string
+  buttonText: string
+  list: IFilterItem[];
+  selected: IFilterDropdownValue;
+  buttonSize?: TypeButtonSizes
+  disabled?: boolean
+  optionType?: IInputOptionsType
+  isLoading?: boolean
+  loadingText?: string
+  searchPlaceholder?: string
+  maxDisplayedItems?: number
+  searchResultText?: string
+  onSelect: (newSelection: IFilterDropdownValue) => void;
+}
+
+const FilterDropdown: React.FC<IFilterDropdown> = ({
+  buttonIcon,
+  buttonText,
+  list,
+  selected = null,
+  buttonSize,
+  disabled,
+  isLoading = false,
+  loadingText,
+  optionType = 'text',
+  searchPlaceholder,
+  maxDisplayedItems = 5,
+  searchResultText = 'Showing [VISIBLE] of [TOTAL]',
+  onSelect = () => { },
+}) => {
+
+  const [visibleList, setVisibleList] = useState(getVisibleList(list, maxDisplayedItems, selected));
+  const [searchText, setSearchText] = useState<string>('');
+  const [contentState, setContentState] = useState<IDropOpen>({
+    isOpen: false,
+    position: 'bottom-right',
+  });
+  const buttonWrapperRef = useRef<HTMLDivElement>(null);
+  const mainRef = useRef<HTMLDivElement>(null);
+
+  const handleToggleOpen = useCallback(() => {
+    if (!buttonWrapperRef.current) { return; }
+
+    const buttonRect = buttonWrapperRef.current?.getBoundingClientRect();
+    if (!buttonRect) { return; }
+
+    const position: IOpenPos = getDropPosition(buttonRect);
+
+    setContentState((prev) => {
+      const isOpen = !prev.isOpen;
+      return { ...prev, isOpen, position }
+    });
+  }, [buttonWrapperRef]);
+
+  const handleClose = useCallback(() => {
+    setContentState((prev) => {
+      const isOpen = false;
+      return { ...prev, isOpen };
+    })
+
+    setSearchText('');
+    setVisibleList(getVisibleList(list, maxDisplayedItems, selected));
+  }, [list, maxDisplayedItems, selected]);
+
+  useClickOutside(mainRef, handleClose);
```

Comment:
This is a little lost here amongst the functions. Can you move it up to under the state or perhaps down to next to the useEffects?

Reply:
I will move it along with handleClose to the top, I can't put it above that function because  uses that function. It can't be use before being declared.